src/neuralnet/neuralnet.cpp

```diff
@@ -63,3 +64,36 @@ INeuralNetPtr NN::GetInstance()
 {
     return instance;
 }
+
+// TODO: perhaps we should give back a smart pointer here. Clients shouldn't
+// really hold onto this or pass it around, though...
+Whitelist& NN::GetWhitelist()
+{
+    return whitelist;
+}
+
+bool NN::AddContract(
+    const std::string& type,
+    const std::string& key,
+    const std::string& value,
+    const int64_t& timestamp)
+{
+    if (type == "project" || type == "projectmapping") {
```

Comment:
what is projectmapping``?

Reply:
Looks like an old contract type name used before "project"... I didn't see any instances of that one on testnet... need to scan mainnet--if it doesn't exist there either, we can remove it.

Reply:
Yep. Carried over from the old code.... probably vestigial.